src/App.tsx

```diff
@@ -15,36 +17,85 @@
   'Garlic',
 ];
 
+const SORT_BY_ALPH: number = 1;
+const SORT_BY_LENGTH: number = 2;
+
+const sortGoods = (goods: string[], key: number): string[] => {
+  switch (key) {
+    case SORT_BY_ALPH:
+      return goods.sort((a, b) => a.localeCompare(b));
+    case SORT_BY_LENGTH:
+      return goods.sort((a, b) => a.length - b.length);
+    default:
+      return goods;
+  }
+};
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState(0);
+  const [toReverse, setToReverse] = useState(false);
+
+  const goods = sortGoods(goodsFromServer.slice(), sortBy);
+
+  if (toReverse) {
+    goods.reverse();
+  }
```

Comment:
[CODE KNOWLEDGE] - It's better to use `Array.prototype.reverse()` method directly when setting the state instead of calling it in the component body. It will prevent unnecessary re-rendering of the component.